app/routes.go

```diff
@@ -3,18 +3,35 @@ package app
 import (
 	"github.com/gorilla/mux"
 	"github.com/ilie20088/go-web-app-boilerplate/app/controllers"
+	"github.com/justinas/alice"
 )
 
+type PublicRouterFactory struct {
+	healthController *controllers.HealthController
+}
+
+func NewPublicRouterFactory() *PublicRouterFactory {
```

Comment:
Is there any need in an explicit factory type? IMHO, we are creating too many thin layers that aren't that useful and add confusion.